webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -348,8 +268,6 @@ class PolicyWriterServiceImpl(
       pathsInfo        =  configAndPaths.map { _.paths }
       templates        <- readTemplateFromFileSystem(techniqueIds)
       resources        <- readResourcesFromFileSystem(techniqueIds)
-      // Clearing cache
-      _                 = fillTemplates.clearCache
       readTemplateTime2 = DateTime.now.getMillis
       readTemplateDur   = readTemplateTime2 - readTemplateTime1
       _                 = policyLogger.debug(s"Paths computed and templates read in ${readTemplateDur} ms")
```

Comment:
this becomes costly
[2019-05-24 11:43:27] DEBUG policy.generation - Paths computed and templates read in 481 ms
per node
rather than per generation

Reply:
yes, per node I have 
[2019-05-24 11:59:28] DEBUG com.normation.rudder.services.policies.write.PolicyWriterServiceImpl - 48 promises templates read in 104 ms
[2019-05-24 11:59:28] DEBUG com.normation.rudder.services.policies.write.PolicyWriterServiceImpl - 14 techniques resources read in 46 ms

we know which nodes are target, so we could compute 
configAndPaths   <- calculatePathsForNodeConfigurations(interestingNodeConfigs, rootNodeId, allNodeConfigs, newPostfix, backupPostfix)
      pathsInfo        =  configAndPaths.map { _.paths }
      templates        <- readTemplateFromFileSystem(techniqueIds)
      resources        <- readResourcesFromFileSystem(techniqueIds)
      // Clearing cache
      _                 = fillTemplates.clearCache
      readTemplateTime2 = DateTime.now.getMillis
      readTemplateDur   = readTemplateTime2 - readTemplateTime1
      _                 = policyLogger.debug(s"Paths computed and templates read in ${readTemplateDur} ms")
before, only once and pass the pathsInfo, templates & resources

Reply:
I put the template parsing out of the node loop. It seems to lead to massive improvment. I also cleaned logs. 
Would you mind testing a war from that branch on load to see how it compare to previous version?